### ISSUE TITLE
Result range hit re-rolls not working

### ISSUE THREAD

**Issue by teddyzetterlund** [dixhuit/unitcrunch-issue-tracker#25]
Tried to keep it simple for the big report.

1. Added a ranged weapon with sustained 2
2. Added re-roll hits of 1–5 to the unit.
3. It returns undefined for the range and doesn’t affect the results.

![IMG_2805](https://github.com/dixhuit/unitcrunch-issue-tracker/assets/1955/930c140c-cb21-4d1a-8661-a791bb46a5ff)

I’ve tried it both on my Mac and iPhone.

**Comment by dixhuit** [comment]
Recreated, thanks. Will try to get this fixed in a patch release.

**Comment by dixhuit** [comment]
I'm struggling to recreate this. Are you able to recreate this issue consistently and are you able to provide any more detail? E.g. did you use any preset modifiers throughout the steps you've listed, if so, for which modifiers?

Please could you recreate the issue if you can and then submit the form at https://www.unitcrunch.com/send-debug-data so I can recreate what you're seeing? Thanks!

**Comment by teddyzetterlund** [comment]
Yep. No problems. I haven't tried any other scenarios really. This is what I did again in an incognito Safari window:
1. Visit unitcrunch.com and accept the welcome message
2. Choose Marines as attacker and defender
3. Add new Attacker: "Immortals"
4. Give them "10 Tesla" A2 S5 AP0 1D" and the preset Sustained Hits 2
5. Give the unit hit re-rolls for the result range 1-5

<img width="686" alt="Screenshot 2023-12-10 at 06 33 04" src="https://github.com/dixhuit/unitcrunch-issue-tracker/assets/1955/fb3ab0c0-b4de-4397-b5c8-45718079762a">

<img width="686" alt="Screenshot 2023-12-10 at 06 34 43" src="https://github.com/dixhuit/unitcrunch-issue-tracker/assets/1955/d8623775-9bdd-4463-8d4f-1b720880330a">

**Comment by teddyzetterlund** [comment]
Same results with Google Chrome doing the same procedure.

**Comment by teddyzetterlund** [comment]
Used the form, ref: pso78efuZQvjabtqOotK

**Comment by dixhuit** [comment]
> Same results with Google Chrome doing the same procedure.

I doubt this is browser related but thanks.

Thanks for all the extra info, I'll have another look at this tonight.

**Comment by dixhuit** [comment]
Weird. I can see a malformed modifier if I load the data that you've sent me and this would absolutely explain the "undefined" that we're seeing in the modifier description as well as why the re-roll is having no effect. But no matter what I try, I can't recreate the issue myself by following the your steps.

That's a bummer as I'm convinced that the problem is created during the creation of the profile, specifically when the re-roll profile ability is created.

> 5. Give the unit hit re-rolls for the result range 1-5

If you have time, would you mind running me through _exactly_ what you're doing during this step? Sorry!

**Comment by teddyzetterlund** [comment]
Maybe this screen recording helps. Here I still get the undefined, although the effect looks like it might take place. I tried again by setting the sustained to 6 just to more clearly see a difference. At that point it got stuck and made no difference if I included it or not in the attacking profile

https://github.com/dixhuit/unitcrunch-issue-tracker/assets/1955/2476aecb-bbbc-43c5-9cbc-f6508d4e1792

**Comment by dixhuit** [comment]
That's very helpful, thank you. I can now recreate the issue. Should have a fix very soon!

**Comment by teddyzetterlund** [comment]
Glad it helped! Will do video recordings in the future. Thanks for being
quick on this one.

**Comment by dixhuit** [comment]
The video was super-helpful but it was actually me being a muppet in the end and not paying enough attention while trying to recreate the bug. Happens sometimes - thanks for your patience!

I've found the issue and have a fix. Am just investigating a potential wider issue that's related as I may want to include a fix for that too.

**Comment by dixhuit** [comment]
Have just released v0.59.1 which resolves this issue. Thanks again for your help!
